### ISSUE TITLE
Command not support: "flutter:waitForFirstFrame"

### ISSUE THREAD

**Issue by JessWeston** [truongsinh/appium-flutter-driver#146]
Apologies for re-raising this issue since it has already been raised: [#28](https://github.com/truongsinh/appium-flutter-driver/issues/28) however that thread seems to have gone stale waiting for a response from the original raiser.

I am trying to run the example project here in this repo using the Java client. However the test fails as soon as it interacts with Flutter in the Appium Server. I have compiled the Kotlin Finder and I am using the example Flutter app under test.

My environment is:
Intel MacOS BigSur
Intellij 2020.3
Emulator: Pixel 3 API 29
Appium 1.20.2
Flutter 2.0.2
Dart 2.12.1

`    capabilities.setCapability(MobileCapabilityType.PLATFORM_NAME, "Android");
    capabilities.setCapability(MobileCapabilityType.PLATFORM_VERSION, "10");
    capabilities.setCapability(MobileCapabilityType.DEVICE_NAME, "emulator-5554");
    capabilities.setCapability(MobileCapabilityType.AUTOMATION_NAME, "Flutter");
`

enableFlutterDriversExtension() is called within the example App.

The first run fails instantly on calling "flutter:waitForFirstFrame" subsequent runs fail after waiting 300 seconds, just like in the #28 issue.

My Appium Server logs: [First Run](https://gist.github.com/JessWeston/471137b57325462381f112834e03c3c0), [Second Run](https://gist.github.com/JessWeston/e45c881ae5c499b0e13084044f7c83d6)

Intellij Run logs: [First Run](https://gist.github.com/JessWeston/eb61f8b0ff269e866917318a5271755c), [Second Run](https://gist.github.com/JessWeston/adf884ed3db52257a55a5645e6265678)

Is there anything wrong with my environment?

**Comment by JessWeston** [comment]
This was caused by issue [#15104](https://github.com/appium/appium/pull/15104) in Appium. After modifying the dependency manually I was able to force Appium to use version 0.0.28 instead of 0.0.25 which did not contain the waitForFirstFrame call that was failing.

**Comment by doc-rj-celltrak** [comment]
@JessWeston Thanks for this information. I'm running into the same problem, but I'm new to Appium. Can you show me how you modified the dependency to pick up version 0.0.28?

**Comment by doc-rj-celltrak** [comment]
@JessWeston This is the pom file I'm using for appium-flutter-finder. I don't see a dependency for appium-flutter-driver, but I do see the appium java-client in there.

```
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
  <modelVersion>4.0.0</modelVersion>
  <groupId>pro.truongsinh</groupId>
  <artifactId>appium-flutter-finder</artifactId>
  <version>0.0.4</version>
  <dependencies>
    <dependency>
      <groupId>org.jetbrains.kotlin</groupId>
      <artifactId>kotlin-stdlib</artifactId>
      <version>1.3.40</version>
      <scope>runtime</scope>
    </dependency>
    <dependency>
      <groupId>org.jetbrains.kotlinx</groupId>
      <artifactId>kotlinx-serialization-runtime</artifactId>
      <version>0.11.1</version>
      <scope>runtime</scope>
    </dependency>
    <dependency>
      <groupId>io.appium</groupId>
      <artifactId>java-client</artifactId>
      <version>2.1.0</version>
      <scope>runtime</scope>
    </dependency>
  </dependencies>
</project>
```

**Comment by JessWeston** [comment]
@doc-rj-celltrak its more of a node thing than a maven thing. This is the dependency that the Appium node package downloads for appium-flutter-driver. I'm on MacOS to the following will show which version my Appium is using:
`cat /Applications/Appium.app/Contents/Resources/app/node_modules/appium/node_modules/appium-flutter-driver/package.json`
This also assumes the Appium desktop install, path's will differ if you installed Appium using npm. To see what your appium is trying to pull use:
`grep appium-flutter-driver /Applications/Appium.app/Contents/Resources/app/node_modules/appium/package.json`
Mine was showing:
`"appium-flutter-driver": "^0",`
Here's where the issue above come in, because ^0 should pull the latest main version 0 in this case 0.0.28, however it is not. So I changed my:
`"appium-flutter-driver": "^0",`
to:
`"appium-flutter-driver": "0.0.28",` in my appium/package.json
Then in order to get appium to pull 0.0.28 and use it you will need to:
`cd /Applications/Appium.app/Contents/Resources/app/node_modules/appium
sudo npm i --unsafe-perm=true --allow-root`

If you're running appium from the commandline then:
`which appium` will tell you where to look, but `bin/appium` is the executable and you want the package.json, which lives in `lib/node_modules/appium`

**Comment by doc-rj-celltrak** [comment]
Thanks for your help @JessWeston. I'm running appium in the Bitrise cloud, but I'll definitely try this locally first. Much appreciated!

**Comment by doc-rj-celltrak** [comment]
@JessWeston thanks again. This worked locally. For what it's worth, and for anyone who needs automation, I found the npe tool nice for editing package.json.

cd /usr/local/lib/node_modules/appium
npm install -g npe
npe dependencies.appium-flutter-driver 0.0.28

I did get a weird error when re-installing appium with --unsafe-perm=true, but it didn't seem to make a difference...

```
> appium@1.20.2 prepare
> gulp prepublish

[11:45:38] No gulpfile found
npm ERR! code 1
npm ERR! path /usr/local/lib/node_modules/appium
npm ERR! command failed
npm ERR! command sh -c gulp prepublish
```
